runtime-decompiler/src/main/java/org/jrd/backend/data/ArchiveManager.java

```diff
@@ -70,8 +84,16 @@ private boolean findClazz(ZipInputStream zis, String clazz) throws IOException {
         return false;
     }
 
-    public static boolean shouldOpen(ZipInputStream zis) throws IOException {
-        return zis.getNextEntry() != null;
+    /**
+     * Determines whether this file can be opened with ZipInputStream
+     * @param n Name of the file
+     * @return Whether file can opened with ZipInputStream
+     */
+    public static boolean shouldOpen(String n) throws IOException {
+        /* This way has been selected as there's no other "easier" way of determining if it is an archive.
+         * If you wish to add a format, feel free to PR */
```

Comment:
pls make this configurable. (no, PR is not configuration :) )

Reply:
Also add a comment about our initial approach  -vith openng and closing stream and why not to use it.

Reply:
Do you want to add a field to the GUI to add extensions?

Reply:
nope! Just a possibility to pass settings in.